HTTP Readiness

hdonnay opened the pull request:
This change (based on #1111) has the http server start serving requests immediately, and start serving the API after initialization is done.

hdonnay commented:
Going to rework claircore's init and then circle back to this.

ldelossa commented:
how does this PR compare with https://github.com/quay/claircore/pull/300 

doesn't #300 obsolete this?

hdonnay commented:
We'll still need to do something in clair's http layers to make use of the information exposed in that change.

ldelossa commented:
Gotchya, was expecting a new PR, but doesn't matter if this one is used.

alecmerdler commented:
@hdonnay Let me know if you need help testing this. This is going to speed up TNG Operator's reconciliation of `QuayRegistry` objects so much faster (Clair updaters take 5+ min to complete, blocking Quay startup).

hdonnay committed:
matcher: add Initialized method

This exposes an underlying change in the claircore packages.

Signed-off-by: Hank Donnay <hdonnay@redhat.com>

hdonnay committed:
httptransport: return Accepted when not ready

This has the HTTP layer return Accepted (202) when the underlying
service reports it's not initialized.

Signed-off-by: Hank Donnay <hdonnay@redhat.com>

hdonnay commented:
This has been reworked and rebased in face of claircore changes.

ldelossa reviewed (changes requested):
Possible to write a integration test for this in some way? Ensuring you see the appropriate status codes change at the appropriate times?

alecmerdler commented:
@hdonnay Would love to see this merged. Is there any way I can help test?

ldelossa commented:
From my point of view, it just needs a test that confirms the http status flips correctly when initialized

hdonnay commented:
I'll jump on a test for the handler. To test it live, you should just need to run it with no updaters (`$.updaters.sets = []`) against an empty database.

ldelossa reviewed (dismissed):
@hdonnay approved, needs rebase, ping me when done for re-ack.
